### PR TITLE
lunar-client: migrate to by-name and add updateScript

### PR DESCRIPTION
**Pull request by surfaceflinger**
## Description of changes

as in title.
also please lmk if this script will work with r-ryantm bot :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 263015` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lunar-client</li>
  </ul>
</details>

Hash & fetched binary works. Update script tested by setting version to `3.0.0` and running it, which correctly reset everything to the values in this PR. LGTM.

Was trying my hardest to see if `update-source-version` could be used instead of `sed`ing around in the package file, but it can't handle `appimageTools.wrapType2` -> `appimageTools.wrapAppImage` -> `buildFHSEnv` not setting a `version` :slightly_frowning_face:.

---

> also please lmk if this script will work with r-ryantm bot :)

If it works for `nix-shell maintainers/scripts/update.nix --argstr package lunar-client`, then it *should* work for the update bot.

**Comment by surfaceflinger** [comment]
> Was trying my hardest to see if `update-source-version` could be used instead of `sed`ing around in the package file, but it can't handle `appimageTools.wrapType2` -> `appimageTools.wrapAppImage` -> `buildFHSEnv` not setting a `version` 🙁.

yeah, I spent quite lots of time trying that already

**Comment by OPNA2608** [comment]
(I guess this package is also lacking `meta.mainProgram`, but again not a deal breaker for this PR)

File: pkgs/by-name/lu/lunar-client/update.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+set -eu -o pipefail
+
+target="$(dirname "$(readlink -f "$0")")/package.nix"
+host="https://launcherupdates.lunarclientcdn.com"
+metadata=$(curl "$host/latest-linux.yml")
+version=$(echo "$metadata" | yq .version -r)
+sha512=$(echo "$metadata" | yq .sha512 -r)
+
+sed -i "s@version = .*;@version = \"$version\";@g" "$target"
+sed -i "s@hash.* = .*;@hash = \"sha512-$sha512\";@g" "$target"
```
Comment (pbsds) [inline pkgs/by-name/lu/lunar-client/update.sh:12]: Concider looking into `common-updater-scripts` instead of using sed

**Review by pbsds — COMMENTED** [review]
You can test the updater script with `nix-update lunar-client -u`

**Pull request merged**